client/src/components/TopNav.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import axios from 'axios';
+import DropDown from './DropDown';
+import Friends from './Friends';
+import Locations from './Locations'
+const Styled = require('./Styles');
+
+// tried to implement styles in a separate file but it isn't working because it says it isn't "returning a string"
+
+const TopNav = (props) => {
+
+  const [text, setText] = useState('');
+  const [manageFriends, setManageFriends] = useState(false);
+
+  const handleSubmit = (e) => {
+    let post = {};
+    e.preventDefault();
+    let updateDB = async () => {
+      let newLocations = [...props.locations, text];
+      console.log('new Locations: ', newLocations)
+      props.setLocations(newLocations);
+      props.setCurrentLocation(text)
+      return newLocations
+    }
+    updateDB()
+    .then((locations) => {
+      post = { locations, 'userName': props.userName, 'currentLocation': props.currentLocation },
+      console.log('this is post in the then: ', post)
+      axios.post('/api/updateLocations', post)
+    }
+    )
+    setText('');
+  }
+
+  const handleRemove = (e) => {
+    let post = {};
+    e.preventDefault();
+    let updateDB = async () => {
+      const removeLocation = () => {
+        let array = [];
+        for (let i = 0; i < props.locations.length; i++) {
+          if (props.locations[i] === props.currentLocation) { continue; }
+          else {array.push(props.locations[i])}
+        }
+        return array;
```

Comment:
This seems like a common piece of logic that's been duplicated from somewhere else. Might be nice to refactor it into some kind of utility method. (Also sort of seems like something that would come as part of lodash or other similar libraries).